core/src/main/java/com/sequenceiq/cloudbreak/service/rdsconfig/RdsConfigService.java

```
@@ -159,4 +122,56 @@ private void setStatusToDeleted(RDSConfig rdsConfig) {
         rdsConfig.setStatus(ResourceStatus.DEFAULT_DELETED);
         rdsConfigRepository.save(rdsConfig);
     }
+
+    public Set<RDSConfig> findAllByOrganizationId(Long organizationId) {
+        return rdsConfigRepository.findAllByOrganizationId(organizationId);
+    }
+
+    @Override
+    protected OrganizationResourceRepository<RDSConfig, Long> repository() {
+        return rdsConfigRepository;
+    }
+
+    @Override
+    protected OrganizationResource resource() {
+        return OrganizationResource.RDS;
+    }
+
+    @Override
+    protected void prepareDeletion(RDSConfig resource) {
+        checkRdsConfigNotAssociated(resource);
+        if (!ResourceStatus.USER_MANAGED.equals(resource.getStatus())) {
+            setStatusToDeleted(resource);
+            throw new BadRequestException(String.format("RDS config '%s' is not usr managed", resource.getName()));
+        }
+    }
+
+    @Override
+    protected void prepareCreation(RDSConfig resource) {
+    }
+
+    public String testRdsConnection(String existingRDSConfigName, RDSConfig rdsConfig, Organization organization) {
+        if (existingRDSConfigName != null) {
+            try {
+                RDSConfig config = getByNameForOrganization(existingRDSConfigName, organization);
+                return testRDSConnectivity(config);
+            } catch (AccessDeniedException | NotFoundException e) {
+                return "not found or access is denied";
+            }
+        } else {
+            return testRDSConnectivity(rdsConfig);
+        }
+    }
+
+    private String testRDSConnectivity(RDSConfig rdsConfig) {
+        try {
+            if (rdsConfig == null) {
+                return "not found";
```

access denied, but maybe better to throw the exception